sound/soc/intel/boards/sof_cirrus_common.c

```
@@ -168,11 +168,16 @@ static int cs35l41_compute_codec_conf(void)
 			continue;
 		}
 		physdev = get_device(acpi_get_first_physical_node(adev));
+		acpi_dev_put(adev);
+		if (!physdev) {
+			pr_devel("Cannot find physical node for HID %s UID %u (%s)\n", CS35L41_HID,
+					uid, cs35l41_name_prefixes[uid]);
+			continue;
```

return 0?

With the continue, you could hypothetically fail on the last device and still report a non-zero set of codecs.

---

yes but asoc properly handles this failure if its trying to register extra nodes, do we really want to stop registering the rest of the nodes?

---

I was referring to the note in the comment where the expected value is 0, 2, 4.

If there is ONE error, you can end-up with an off configuration, e.g. one codec. Better to just signal to the higher level that the configuration isn't quite right. Fail big and fail early, eh?

---

fair enough, if we know the firmware says its there but we are having trouble finding it that is a pretty clear